internal/plugincommon/chain_support.go

```diff
@@ -87,7 +87,7 @@ func (c ccipChainSupport) SupportedChains(oracleID commontypes.OracleID) (mapset
 		return mapset.NewSet[cciptypes.ChainSelector](), fmt.Errorf("error getting supported chains: %w", err)
 	}
 
-	return supportedChains, nil
+	return supportedChains.Clone(), nil
```

Comment:
Return a clone from supported chains. Otherwise any operations on the returned set will affect the original set